Tools/LyTestTools/ly_test_tools/o3de/editor_test_utils.py

```diff
@@ -151,3 +152,42 @@ def retrieve_last_run_test_index_from_output(test_spec_list: list[EditorTestBase
         else:
             index += 1
     return index
+
+def save_failed_asset_joblogs(workspace: AbstractWorkspace) -> None:
+    """
+    Checks all asset logs in the JobLogs directory to see if the asset has any warnings or errors. If so, the asset is
+    saved via ArtifactManager.
+
+    :param workspace: The AbstractWorkspace to access the JobLogs path
+    :return: None
+    """
+    # Tuples are in the form [this directory, [child directories], [files in this directory]]
```

Comment:
Not necessary to document os.walk here, unless something uncommon is being done and needs explanation